### DISCUSSION TITLE
Is it possible to iterate over output of previous node?

### DISCUSSION THREAD

**Discussion by pbonito** [microsoft/promptflow#3566] [Q&A]
Hi team,
we would like call recursively a custom_llm node based on the output of a python node that reads file from a dir and emits the contents for llm analysis. Next step should get output from llm and save as output.
Is this possible or we should put the iteration logic outside promptflow?
Thanks

**Comment by brynn-code** [comment]
Thanks for reaching us, for now promptflow does not have 'loop' operator, but you can still achieve this in two ways.
1. Use flex flow
Flex flow doesn't have 'node' concept, we could define a python function or a python class as flow directly, which means you could do any logic within the python function. Here is the documentation about flex flow: https://microsoft.github.io/promptflow/how-to-guides/develop-a-flex-flow/function-based-flow.html and examples: https://microsoft.github.io/promptflow/tutorials/flex-flow-quickstart.html

2. Use dag flow with flow as function feature (Internal, not recommended)
If you would like to keep the 'node' concept, you could still achieve this by calling another flow in the python tool with the help of 'call flow as a function' feature. Here is the documentation about this feature: https://microsoft.github.io/promptflow/tutorials/flow-as-function.html#example2-load-flow-as-a-function-with-in-memory-connection-override, and when using in python tool (flow node), it will be something like:
```python
f = load_flow(Path(__file__).parent / "my_llm_flow")
@tool
def execute_flow_tool(connection: AzureOpenAIConnection):
    f.context.connections = {"classify_with_llm": {"connection": connection}}
    f(url=sample_url)   # Execute flow
```
